src/pybind/mgr/rook/module.py

```diff
@@ -538,7 +560,23 @@ def apply_drivegroups(self, specs: List[DriveGroupSpec]) -> OrchResult[List[str]
                 raise RuntimeError("Rook cluster configuration does not "
                                 "support OSD creation.")
             result_list.append(self.rook_cluster.add_osds(drive_group, matching_hosts))
-        return OrchResult(result_list)
+        return result_list
+
+    def _load_drive_groups(self) -> None:
+        stored_drive_group = self.get_store("drive_group_map")
+        self._drive_group_map: Dict[str, DriveGroupSpec] = {}
+        if stored_drive_group:
+            for name, dg in json.loads(stored_drive_group).items():
+                try:
+                    self._drive_group_map[name] = DriveGroupSpec.from_json(dg)
+                except ValueError as e:
+                    self.log.error(f'Failed to load drive group {name} ({dg}): {e}')
+
+    def _save_drive_groups(self) -> None:
+        json_drive_group_map = {
+            name: dg.to_json() for name, dg in self._drive_group_map.items()
+        }
+        self.set_store("drive_group_map", json.dumps(json_drive_group_map))
```

Comment:
What I don't like here is that we're hiding those drive groups from Rook by storing them in the MONs. This feels so wrong to me. We've made the same mistake with OCS where managing the Rook CRs from the OCS CRs making is somewhat hard to even enable the Dashboard in an OCS context. And I feel we're making the same mistake all over again here.

Reply:
Rook doesn't understand drive groups--they're a ceph feature.  I don't see how we can not "hide" them from rook.

We could store them in k8s somewhere (ConfigMap?) I guess, but that doesn't actually accomplish anything except make the code more complex.